Source/MQTTnet/MqttClient.cs

```diff
@@ -947,7 +971,15 @@ async Task TryProcessReceivedPacket(MqttPacket packet, CancellationToken cancell
             switch (packet)
             {
                 case MqttPublishPacket publishPacket:
-                    EnqueueReceivedPublishPacket(publishPacket);
+                    if (Options.ReceivedApplicationMessageQueueable)
+                    {
+                        // publishPacket must be copied
+                        EnqueueReceivedPublishPacket(publishPacket.Clone());
```

Comment:
The publishPacket here must be cloned, otherwise problems will occur after replacing it with IMqttClientAdapterFactory of mqttnet.asp.
We can consider using Pool memory for MQTTnet.Adapter.MqttChannelAdapter to avoid unnecessary cloning of publishPacket.